### ISSUE TITLE
Unecessary `txa` in simple copy

### ISSUE THREAD

**Issue by sasq64** [llvm-mos/llvm-mos#410]
```c
    for(unsigned char i = len; i>0; i--) {
        (dst-1)[i] = (src-1)[i];
    }
```

```asm
.LBB0_1:
        lda     8191,x
        sta     32767,x
        dex
        txa
        bne     .LBB0_1
```

`txa` instruction has no effect.

https://godbolt.org/z/TenT7ohzs

**Comment by johnwbyrd** [comment]
@mysterymath will know the answer to this one, but I suspect that this is not considered for optimization because llvm-mos does not track the state of N and Z flags.

**Comment by mysterymath** [comment]
There's a bit of code that runs at the very end and lowers N and Z flag generation, looking for available expressions that already generate them. Not sure why that isn't triggering here, but it's definitely within the scope of that pass.

**Comment by asiekierka** [comment]
I built the latest version of LLVM-MOS, and can no longer reproduce this. Was this accidentally fixed?

**Comment by mysterymath** [comment]
> I built the latest version of LLVM-MOS, and can no longer reproduce this. Was this accidentally fixed?

I can confirm this as well; may have happened as part of the CmpBr work. Closing.
